### ISSUE TITLE
External Links in Nested Content removed when I Restore content on Umbraco Cloud

### ISSUE THREAD

**Issue by jeannieSoS** [umbraco/Umbraco.Cloud.Issues#179]
I have a couple of starter projects on Umbraco Cloud.  I have in each project a document type that includes a Nested Content data type. The Nested Content data type includes a Multi Url Picker data type (set to allow just 1 link to be added). This has been working perfectly until yesterday. Now trying to Restore content from Live to local or Development, removes any external links. 
 
To reproduce, go to the Content section, select node and Restore partial. Select the Live environment and restore. The external links are removed. Over the last couple of days when I try to do a Partial restore to pull content down from the Live environments any external links in Nested Content are not pulled down and in fact are removed from content on Local and Development.

I am seeing this issue on Umbraco Cloud version: Umbraco 8.5.4

This happens on 2 completely separate projects. I've tried using Chrome Version 78.0.3904.97 and Microsoft Edge 42.17134.1098.0 browsers.

**Comment by jmayntzhusen** [comment]
@jeannieSoS 

Just to confirm, is this with Deploy 3.4.0? 🙂

**Comment by jeannieSoS** [comment]
No it's on 3.3.1 still. I don't have an option to upgrade

**Comment by jmayntzhusen** [comment]
That's unfortunate, have had a couple reports of that so we are looking into why the upgrade button doesn't appear. In the meantime you can install it manually if you need.

1. Download it here: https://umbraconightlies.blob.core.windows.net/umbraco-deploy-release/UmbracoDeploy.v3.4.0.zip
1. Unzip and drag all 3 folders over to your local site
1. Replace all of your current files **EXCEPT** deploy.settings.config and deploy.config 
1. Test the site runs just fine locally
1. Push to your development env
1. Test if the restore to local works as expected

If it is still a problem with Deploy 3.4.0 we'd like to hear more about your setup, but we have tested based on your initial report with a fresh site where a node had a nested content property editor with a multiurl picker and the links were transferred without issues.

**Comment by jeannieSoS** [comment]
I've run the upgrade against one of my projects and the links are still disappearing unfortunately

**Comment by jeannieSoS** [comment]
I've now run the project manually against 2 projects and the automated way against the other and the external links are removed every time I Restore or Partial Restore content from the Live environment.

**Comment by HalldorLyngmo** [comment]
Hi @jeannieSoS 

I've been unable to reproduce your issue so far on a fresh project running:
- Umbraco v8.5.4
- Deploy v3.4.0
- Umbraco.Deploy.Contrib.Connectors v.3.0.0

My setup is like this. I got a document type that has a nested content property. Then I have another document type which is an element. That one has one property which is a Multi URL Picker and it has a maximum of 1 link like you mentioned earlier in the issue.

On the nodes, I've added links to full URLs and also selected nodes inside the backoffice. Lastly, I linked some media items. Everything has been restored successfully with both a partial and full restore.

Could you clarify what the difference might be in my setup versus yours? Maybe I missed something...

You could also reach out to your support team, reference the issue and we then can get access to one of your projects to investigate further. 🙂

**Comment by jeannieSoS** [comment]
Your test set-up sounds the same. I'm having the same problem across 3 projects. I spoke to Umbraco support initially so I'll get back to them to allow you access.

**Comment by nul800sebastiaan** [comment]
I'll close this for now with the assumption that this is problem specifically to your project. Hope support can find the problem!

**Comment by jeannieSoS** [comment]
I've managed to resolve it by republishing the site, on all 3 of the live sites. I'm still not sure why this happened. Would be good to know.

**Comment by HalldorLyngmo** [comment]
@jeannieSoS 

Unfortunately, we couldn't quite pinpoint where the issue started. We believe it was connected to the previous bugs we had with nested content and deploy/deploy.contrib.

In your project, we looked at the values in the Database for the Multi Url Picker in the nested content. We noticed that it was saved in the wrong format. So publishing the node again changed it to the correct format which then enabled Deploy to handle it like it should when transferring between environments.
